### ISSUE TITLE
`checkClientInfo` fails to require `client.js` when package.json is `"type": "module"`

### ISSUE THREAD

**Issue by jeffsee55** [tinacms/tinacms#3315]
When running `tinacms build` with the latest updates we `require` the client so we can check that the config has been set up with valid credentials. 

```
ExecuteSchemaError: Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/jeffsee/code/scratch/tina-type-module-bug/.tina/__generated__/temp_client/client.js from /Users/jeffsee/code/scratch/tina-type-module-bug/node_modules/@tinacms/cli/dist/index.js not supported.
client.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename client.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/jeffsee/code/scratch/tina-type-module-bug/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).
```

When we moved to Vite 3 it was necessary to deprecate our use of the ESM module, which meant our CLI would now run as commonjs. The problem is that when a `package.json` has `"type": "module"` in it, the generated client will be treated as an ES module, meaning the require will fail. 

A similar issue was fixed by specifying the `config.js` be compiled to `config.cjs`, but this is probably not the best solution for this, as we actually _do_ want it to be an es module when it's consumed by the end user in their app.

I think we have 2 options:
1. Generate 2 clients for now one will go in a temp directory that we clean up and have the file ending `.cjs`. The other will continue to be used as usual.
2. Run the CLI as an ES module, this is probably one of those things we need to do eventually as the ecosystem moves away from CJS.

**Comment by jeffsee55** [comment]
3rd option - We can just grab the API url from the generation and use that for our `checkClientInfo` call

https://github.com/tinacms/tinacms/blob/966bcb9741ce13399b76afb67365501b64326546/packages/@tinacms/cli/src/buildTina/index.ts#L179

**Comment by jamespohalloran** [comment]
Can this be closed now with #3359 being merged?

Side-note, [Microbundle](https://github.com/developit/microbundle) might be handy if we do want to go with Option 1 at some point to generate esm/non-esm bundles

**Comment by jeffsee55** [comment]
Yeah this can be closed. Regarding microbundle, we can do option 1 pretty easily with our current setup, too if we choose to at some point.
